modules/_labs/color-picker/react/lib/ColorPicker.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import {ColorInput} from '@workday/canvas-kit-react-color-picker';
+import {borderRadius, colors, spacing} from '@workday/canvas-kit-react-core';
+import {Label} from '@workday/canvas-kit-react-form-field';
+import {IconButton} from '@workday/canvas-kit-react-button';
+import {checkIcon} from '@workday/canvas-system-icons-web';
+
+import {SwatchBook} from './parts/SwatchBook';
+import {ResetButton} from './parts/ColorReset';
+
+export interface ColorPickerProps {
+  ariaLabel?: string;
+
+  onChange: (color: string) => void;
+  selectedColor?: string;
+  customColors?: string[];
+
+  showCustomInput?: boolean;
```

Comment:
> Nit: Anything with defaults should not be typed as optional

can you expand on this? it has a default because it is optional, so I'm not sure what you mean

Reply:
Even for booleans, we still put them in `defaultProps`. Technically `showCustomInput` has a default value of `undefined`, not `false`. It doesn't make a difference here since you're only checking if it's truthy, but in some cases it can make a difference ([example](https://github.com/Workday/canvas-kit/blob/master/modules/button/react/README.md#toggled-boolean--undefined)). Just good practice 🙂 

In terms of optional typing, [anything in defaultProps should be typed as required, but documented as optional](https://github.com/Workday/canvas-kit/blob/master/API_PATTERN_GUIDELINES.md#default-props). This is a typescript thing.

Let me know if that clarifies it!

Reply:
I don't think making these components class-based just to be able to use `defaultProps` is necessary? I'll go ahead and assign defaults while destructuring the props instead.

Reply:
It's still possible to add `ColorPicker.defaultProps`, keeping it a functional component. Defaults during destructuring are also a reasonable alternative.

Referencing some research @NicholasBoll did:

```tsx
// Example 1
// * Typescript understands `optional` is optional externally, but always defined inside render function
// * Storybook Docs can understand the properties and will make a prop table
// * Storybook Docs can understand default values (only non-functions)
// * `Foo.defaultProps` cannot be type-safe without intermediate object
​
interface FooProps {
  optional: string
}
​
export function Foo({ optional }: FooProps) {
  // optional is `string` here. Typescript knows it is always defined
  return <div>{optional}</div>
}
​
// Type checking is not available.
// Can't force to `Partial<FooProps>` without another object
Foo.defaultProps = {
  optional: 'bar',
}
​
// `optional` is not required. Typescript understands `defaultProps` makes it optional
const foo = <Foo />
​
// Example 2
// * `optional` is forced to be optional through interface - default provided by deconstruction
// * Storybook Docs can understand the properties and will make a prop table
// * Storybook Docs do _not_ understand default props
// * Type-checking default values is automatic
​
interface FooProps {
  optional?: string
}
​
export function Foo({ optional = 'bar' }: FooProps) {
  // optional is `string` here. Typescript knows it is always defined
  return <div>{optional}</div>
}
​
// `optional` is not required per interface
const foo = <Foo />
```

I lean towards the first method because the default props are able to be pulled out by the storybook docs add on properly. The lack of type checking when assigning defaultProps isn't a huge deal since it's usually a pretty intentional assignment. 

Using defaults during destructuring is cleaner (and would be my preference), but unfortunately our storybook docs plugin doesn't support it yet 🙁

Reply:
After a bit more conversation on Slack, we would all lean towards the second option and we're going to look into updating storybook docs to support it. Let's stick with your destructured defaults instead 🙂

Reply:
sweet! thanks